src/main/java/core/basesyntax/strategy/MyFileWriterImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.strategy;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class MyFileWriterImpl implements MyFileWriter {
+    public void write(String report, String filePath) {
+        File file = new File(filePath);
+        File parentDir = file.getParentFile();
+        if (parentDir != null && !parentDir.exists()) {
+            parentDir.mkdirs();
+        }
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(file))) {
+            writer.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("can't write data to" + filePath, e);
```

Comment:
Avoid starting exception messages with a lowercase letter and ensure to include a space before the file path in the error message for readability. For example: "Can't write data to " + filePath.